src/ray/core_worker/actor_manager.cc

```diff
@@ -3,19 +3,18 @@
 
 namespace ray {
 
-void ActorManager::PublishCreatedActor(const TaskSpecification &actor_creation_task,
-                                       const rpc::Address &address) {
-  auto actor_id = actor_creation_task.ActorCreationId();
-  auto data = gcs::CreateActorTableData(actor_creation_task, address,
-                                        gcs::ActorTableData::ALIVE, 0);
-  RAY_CHECK_OK(global_actor_table_.Append(JobID::Nil(), actor_id, data, nullptr));
-}
-
 void ActorManager::PublishTerminatedActor(const TaskSpecification &actor_creation_task) {
   auto actor_id = actor_creation_task.ActorCreationId();
   auto data = gcs::CreateActorTableData(actor_creation_task, rpc::Address(),
                                         gcs::ActorTableData::DEAD, 0);
-  RAY_CHECK_OK(global_actor_table_.Append(JobID::Nil(), actor_id, data, nullptr));
+
+  auto update_callback = [actor_id](Status status) {
+    if (!status.ok()) {
+      // Only one node at a time should succeed at creating or updating the actor.
+      RAY_LOG(FATAL) << "Failed to update state to DEAD for actor " << actor_id;
```

Comment:
How about we log error instead?

Reply:
Also, please include `status` in the log message.

Reply:
Is there a specific scenario that you think this could fail?

Reply:
Maybe a race condition with reconstruction logic in the raylet?

Reply:
Raylet only updates actor table after the actor creation task finishes successfully, while actor owner only does it when actor creation task fails and the actor cannot be created. I think there shouldn't be conflicts.

Reply:
Well, it could still happen if we got a spurious RPC failure that was delayed. I am not that worried about this scenario, just pointing out that in the worst case it is plausible.

Reply:
Yep, thanks for the pointers:)